homeassistant/components/watergate/sensor.py

```diff
@@ -0,0 +1,257 @@
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    WatergateSensorEntityDescription(
+        value_fn=lambda data: data.telemetry.pressure if data.telemetry else None,
+        translation_key="water_pressure",
+        key="water_pressure",
+        native_unit_of_measurement=UnitOfPressure.MBAR,
+        device_class=SensorDeviceClass.PRESSURE,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    WatergateSensorEntityDescription(
+        value_fn=lambda data: (data.telemetry.flow / 1000)
+        if (data.telemetry and data.telemetry.flow is not None)
+        else None,
+        translation_key="water_flow_rate",
+        key="water_flow_rate",
+        native_unit_of_measurement=UnitOfVolumeFlowRate.LITERS_PER_MINUTE,
+        device_class=SensorDeviceClass.VOLUME_FLOW_RATE,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    WatergateSensorEntityDescription(
+        value_fn=lambda data: data.state.uptime if data.state else None,
+        translation_key="uptime",
+        key="uptime",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+        native_unit_of_measurement=UnitOfTime.SECONDS,
+        device_class=SensorDeviceClass.DURATION,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    WatergateSensorEntityDescription(
+        value_fn=lambda data: data.state.power_supply if data.state else None,
+        translation_key="power_supply",
+        key="power_supply",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+    ),
+]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: WatergateConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up all entries for Watergate Platform."""
+
+    coordinator = config_entry.runtime_data
+
+    entities: list[SensorEntity] = [
+        SonicSensor(coordinator, description) for description in DESCRIPTIONS
+    ]
+
+    entities.extend([AutoShutOffEventSensor(coordinator)])
+
+    async_add_entities(entities)
+
+
+class SonicSensor(WatergateEntity, SensorEntity):
+    """Define a Sonic Sensor entity."""
+
+    entity_description: WatergateSensorEntityDescription
+
+    def __init__(
+        self,
+        coordinator: WatergateDataCoordinator,
+        entity_description: WatergateSensorEntityDescription,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(coordinator, entity_description.key)
+        self.entity_description = entity_description
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle data update."""
+        self._attr_native_value = self.entity_description.value_fn(
+            self.coordinator.data
+        )
+        self._attr_available = self._attr_native_value is not None
```

Comment:
I think it makes sense to add a proper `available` property (including `super().available)`, because if the coordinator update fails this code is not called and the device stays available.